src/Firebase/FirebaseAuth.js

```diff
@@ -0,0 +1,155 @@
+import { app } from './FirebaseConfig';
+import {
+	getAuth,
+	signOut,
+	GoogleAuthProvider,
+	createUserWithEmailAndPassword,
+	updateProfile,
+	signInWithEmailAndPassword,
+	signInWithPopup,
+} from 'firebase/auth';
+import { addUserToTheDB, getSpecificUser } from './FirebaseFirestore';
+const auth = getAuth(app);
+const defaultImage =
+	'https://firebasestorage.googleapis.com/v0/b/react-testing-firebase-39d5e.appspot.com/o/defaultImages%2Fdefault.jpg?alt=media&token=1f3c0b1d-9d99-406b-b524-f6e278f7c5c7';
+
+const getNewUserObject = (currentUser) => {
+	return {
+		displayName: currentUser.displayName,
+		uid: currentUser.uid,
+		photoURL: currentUser.photoURL,
+		email: currentUser.email,
+		followers: 0,
+		following: 0,
+		posts: 0,
+		bio: '',
+		url: '',
+		backgroundImageUrl: '',
+	};
+};
+const emailSignUpHandler = async (errorHandler, setIsLoading, dispatch, setTextFields, textFields, setAuthUser) => {
+	const { email, password, confirmPassword, name } = textFields;
+	if (name.length < 2) {
+		setTextFields({ ...textFields, nameError: true });
+		errorHandler(true, 'Name must be atleast 2 characters long');
+		return;
+	}
+	if (email.length === 0) {
+		setTextFields({ ...textFields, emailError: true });
+		errorHandler(true, 'Email is required');
+		return;
+	}
+	if (
```

Comment:
If we put input type = email with required attribute it will automatically whether you have typed valid or invalid email.
There's no need to check here